src/xdp-utils.c

```diff
@@ -572,6 +573,62 @@ cleanup_temp_file (void *p)
   g_free (*pp);
 }
 
+static GKeyFile *
+spawn_validator (const char* const* args,
+                 int                fd)
+{
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  g_autoptr(GSubprocessLauncher) launcher = NULL;
+  g_autoptr(GSubprocess) process = NULL;
+  int stdin_fd = -1;
+  int stdout_fd = -1;
+  g_autofree char *stdout_path = NULL;
+
+  stdout_fd = memfd_create("spawn-validator-stdout", 0);
+
+  if (stdout_fd == -1)
+    {
+      int saved_errno;
+
+      saved_errno = errno;
+      g_warning ("Failed to spawn subprocess: memfd_create: %s", g_strerror (saved_errno));
```

Comment:
why isn't this just `g_strerror (errno)`?

Reply:
It’s best practice to save `errno` to a local variable on the next line after calling the function which could fail, as other functions (particularly in the error handling / warning code call chain) could modify `errno`.

Even if that doesn’t actually happen here, this approach makes the code more maintainable (e.g. if someone were to refactor the `g_warning()` call in future and add in some other calls which potentially modify `errno`).

Reply:
This is also what the docs of `g_strerror (errno)` say.